go/logictest/runner.go

```diff
@@ -225,6 +235,30 @@ func skipUntilEndOfRecord(scanner *parser.LineScanner, wr *bufio.Writer) {
 	}
 }
 
+type loggingLock struct {
+	mux *sync.Mutex
+	logged bool
+}
+
+func newLoggingLock() *loggingLock {
+	return &loggingLock{
+		mux: &sync.Mutex{},
+		logged: false,
+	}
+}
+
+func (ll *loggingLock) GetLogged() bool {
```

Comment:
Unfortunately, a separate method for getting and setting the logged property won't work. To prevent a race you must get the lock, check the value, set the value, then release the lock. As you have this now, one goroutine can call GetLogged, see it's false, then another can call GetLogged, see it's false, and then both log (and then both call SetLogged().

I would just eliminate these methods and let callers access the field directly.